Add courses to the repository

statnmap opened the pull request:
The content will be built using an internal ThinkR package called {formation}.
This ensures that the HTML output, datasets, and extra files are stored in the correct place to be added to the ThinkR e-learning platform.  
Each instructor is still able to locally knit each chapter using {xaringan}.  
The content of the course is:  

- courses/C00-package-objectives.en.Rmd: What we will present during this tutorial, and how it will be held
- courses/00-cruz.Rmd: How the e-learning platform works
- courses/C00-package_preambule.en.Rmd: What means package first ?
- courses/C01-package_express_fusen.en.Rmd: Build a package using {fusen}
- courses/C06-pkg_avance_test.Rmd: Go further with unit tests

statnmap committed:
init courses content

statnmap committed:
Add renv and fusen

statnmap committed:
Update Readme

statnmap commented:
I plan to try a GitHub Actions to compile the course with the ThinkR template.

emilyriederer commented on courses/C01-package_express.en.Rmd:
```diff
@@ -0,0 +1,459 @@
+
+In Rstudio : 
+
+File > New project > new directory > R package **using devtools**  
```
Should we use `usethis` to create the directory instead of the IDE?

statnmap replied:
I add the possibility. Thanks.

emilyriederer commented on courses/C01-package_express.en.Rmd:
```diff
@@ -0,0 +1,459 @@
+File > New project > new directory > R package **using devtools**  
+We choose the name of the package (explicit, in lower case) and a folder on the computer  
+
+- Name of the package : "hello
```
^Missing close quotation mark

statnmap replied:
Corrected in the next version

emilyriederer commented on courses/C01-package_express.en.Rmd:
```diff
@@ -0,0 +1,459 @@
+The description of the package is done in the "DESCRIPTION" file.  
+The first fields to fill in :
+
+- Title: quick description
```
I know CRAN has some strict rules around title/description (e.g. don't reuse package name, certain capitalization, etc.) Even though most _analyisis_ packages will not end up on CRAN should we still provide these as the best practices?

statnmap replied:
Yes. Some of these rules can break the check(). Added.

emilyriederer commented on courses/C01-package_express.en.Rmd:
```diff
@@ -0,0 +1,459 @@
+### Step 2 : Description
+
+The description of the package is done in the "DESCRIPTION" file.  
+The first fields to fill in :
```
Do we wish to encourage participants to edit the `DESCRIPTION` directly or via the `usethis` functions? I have no strong preference

emilyriederer commented on courses/C01-package_express.en.Rmd:
```diff
@@ -0,0 +1,459 @@
+
+We create a file to save all our manual operations
+
+- Create a "dev/" folder
```
I see that you use `usethis::use_build_ignore()` to ignore the `dev/` folder. I wonder if this is worth explicitly listing out as a step and explaining that _any_ additional files/directories can be included in a package but those not related to the core package must be ignored?

emilyriederer commented on courses/C01-package_express.en.Rmd:
```diff
@@ -0,0 +1,459 @@
+print("Hello person")
+
+person <- "Seb"
+print(glue::glue("Hello {person}"))
```
When you've taught this previously, does everyone seem to know the `glue` package? I think it is pretty obvious what this code is doing (which is a good thing!), but alternatively we could use `paste()` since everyone is familiar

emilyriederer replied:
However, I do see later that this motivates the use of `importFrom` in the Rd, so perhaps this is strategic! 🙂

statnmap replied:
Yes this is strategic :smile:

emilyriederer commented on courses/C06-pkg_avance_test.Rmd:
```diff
@@ -0,0 +1,247 @@
+
+- Add unit test in your {hello} package to test that `say_hello()` returns a message
+- Check the package
+
```
I wonder if we could/should add anything about how to come up with good test cases? I often think new developers do not think carefully about creating _hypotheses_ for how their code could fail and evaluating actual corner cases

emilyriederer commented on courses/C01-package_express_fusen.en.Rmd:
```diff
@@ -0,0 +1,540 @@
+
+???
+
+- We use the same exact example to rite the unit test directly.  
```
Tiny typo: "rite" -> "write"

emilyriederer commented on courses/C01-package_express_fusen.en.Rmd:
```diff
@@ -0,0 +1,540 @@
+- Use a test set for your examples and tests
+  + `usethis::use_data()`: function to create an internal dataset  
+  
+- Add a version tracking file
```
Should we also mention `usethis::use_version()`? Participants coming from an analysis context might also appreciate advise on when/how often to update different parts of the version number

statnmap replied:
That is a common question yes. Let's add this.

emilyriederer reviewed (commented):
Thank you for sharing -- this looks great! I added a few minor thoughts/comments as I read through just for discussion, but there is nothing I feel strongly about changing. 

I have one broader question about `fusen`: once one "inflates" the package, do they need to continue to develop exclusively within the Rmd or exclusively from the generated files? I am wondering if we should teach any changes within the resulting package file structure or if the two workflows are not compatible.

statnmap committed:
Update fusen presentation in 2 steps

statnmap commented:
I removed the course named "courses/C01-package_express.en.Rmd", I think we can directly use the version with {fusen} (courses/C01-package_express_fusen.en.Rmd)  
I updated its content following your feedbacks and those of my current attendees.  
I removed the course on unit tests, as we proposed to speak about including data in the package. I do not think we will be able to cover both topics.  
Then, I added a course on data, still in French for now, but I will translate it, and add an example of use from {fusen} using the full dev template with `fusen::add_dev_history()`

statnmap committed:
First try on Actions

statnmap committed:
Chapter on data integration

statnmap committed:
Data: add quizz

statnmap committed:
tests and data courses and exo

statnmap committed:
CI: test again renv

statnmap committed:
CI: install as Rscript

statnmap committed:
Rm formation from desc

statnmap committed:
CI: with correct env variable

statnmap committed:
CI: dont forget remotes

statnmap committed:
CI: Ubuntu and renv first

statnmap committed:
CI deploy

statnmap committed:
CI: system deps and rspm

statnmap committed:
CI: syst deps

statnmap committed:
CI: syst deps correction

statnmap committed:
CI with the-ci

statnmap committed:
CI: add repo_name

statnmap committed:
CI: install remotes before

statnmap committed:
deploy differently

statnmap committed:
CI: correct folder

statnmap committed:
CI: back to prev deploy

statnmap committed:
CI: dont ignore html

statnmap committed:
CI: move to public
